sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -297,6 +297,7 @@ case class Literal (value: Any, dataType: DataType) extends LeafExpression {
   override def toString: String = value match {
     case null => "null"
     case binary: Array[Byte] => s"0x" + DatatypeConverter.printHexBinary(binary)
+    case d: ArrayBasedMapData => s"map(${d.toString})"
```

Comment:
Just a question; any reason not to update `ArrayBasedMapData#toString` instead?
https://github.com/apache/spark/blob/cbd3fdea62dab73fc4a96702de8fd1f07722da66/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapData.scala#L35-L37

Reply:
Yes. It's because that class is `MapData` technically, not `Map`.

Reply:
The **static map**'s `Literal` representation happens to be `ArrayBasedMapData` for now, but it can be changed later.